### PR TITLE
Fix to Wrapped Condenser HPWH connections to coil when there is more than one HPWH

### LINKED ISSUE #5127: HPWH wrapped condenser error when more than one in simulation
When there is more than one `WaterHeater:HeatPump:WrappedCondenser` in a single building simulation, the following error occurs:

```
** Severe  ** WaterHeater:HeatPump:WrappedCondenser = HPWH_1:
**   ~~~   ** Heat Pump condenser water inlet node name does not match water heater tank source outlet node name.
**   ~~~   ** Heat pump condenser water inlet and outlet node names = DUMMY CONDENSER INLET HPWH COIL_1 and DUMMY CONDENSER OUTLET HPWH COIL_1
**   ~~~   ** Water heater tank source side inlet and outlet node names      = DUMMY CONDENSER OUTLET HPWH COIL_2 and DUMMY CONDENSER INLET HPWH COIL_2
** Severe  ** WaterHeater:HeatPump:WrappedCondenser = HPWH_1:
**   ~~~   ** Heat Pump condenser water outlet node name does not match water heater tank source inlet node name.
**   ~~~   ** Heat pump condenser water inlet and outlet node names = DUMMY CONDENSER INLET HPWH COIL_1 and DUMMY CONDENSER OUTLET HPWH COIL_1
**   ~~~   ** Water heater tank source side inlet and outlet node names      = DUMMY CONDENSER OUTLET HPWH COIL_2 and DUMMY CONDENSER INLET HPWH COIL_2
**  Fatal  ** Errors found in getting WaterHeater:HeatPump:WrappedCondenser input. Preceding condition causes termination.
```

@jmaguire1

### PR DESCRIPTION
**Pull request by nmerket**
This fixes #5127. With unit test, of course.

### CODE REVIEW

File: src/EnergyPlus/WaterThermalTanks.cc [hunk 1]
```diff
@@ -3612,7 +3612,7 @@ namespace WaterThermalTanks {
 								ShowContinueError( "Please leave the source side inlet and outlet fields blank." );
 								ErrorsFound = true;
 							} else {
-								WaterHeaterSaveNodes &HPWHNodeNames = HPWHSaveNodeNames( NumHeatPumpWaterHeater );
+								WaterHeaterSaveNodes &HPWHNodeNames = HPWHSaveNodeNames( HPWaterHeaterNum );
```
Comment (nmerket) [inline src/EnergyPlus/WaterThermalTanks.cc:3615]: Was looking up the water heater node names using the variable for "total number of heat pump water heaters" when it should have been looking it up by the variable for "current heat pump water heater number". Worked fine when there was only one HPWH in the idf (`NumHeatPumpWaterHeater == HPWaterHeaterNum`).

**Comment by nmerket** [comment]
Who can review this? @Myoldmopar @mbadams5  It's a pretty easy/dumb fix. 

The most interesting part is the unit test.

**Comment by Myoldmopar** [comment]
> hpwh dummynodes

Oh man, @EnergyArchmage @mjwitte I just had terrible visions of some [super-special tank nodes](https://github.com/NREL/EnergyPlusArchive/commit/e8fa020b8e83730c8e06863655004a7064c2cf87).

**Comment by Myoldmopar** [comment]
What?  That's an awesome unit test.  And the unit test passes to verify that the GetInput is working for multiple tanks.  I'll merge when it is all green.

**Comment by Myoldmopar** [comment]
@nmerket, can you edit the name of the pull request to be something more digestible for a user looking at the changelog?

**Pull request merged**